### PR TITLE
feat: add nacos example

### PR DESCRIPTION
**Pull request by dengWuuu**
#### What type of PR is this?
feat

#### What this PR does / why we need it (English/Chinese):
en: A more detailed case study according to the cloudWeGo documentation, adding a post request with json
zh: 按照官方文档添加了更详细的案例，增加一个post请求且参数是json在服务发现中的使用

#### Which issue(s) this PR fixes:
https://github.com/cloudwego/hertz/issues/485

### CODE REVIEW

File: nacos/examples/custom_config/client/main.go [hunk 1]
```diff
@@ -16,22 +16,28 @@ package main
 
 import (
 	"context"
-
+	"encoding/json"
+	"fmt"
```
Comment (L2ncE) [inline nacos/examples/custom_config/client/main.go:20]: Please use two-stage like below
```go
"context"
"encoding/json"

"github.com/cloudwego/hertz/pkg/app/middlewares/client/sd"
"github.com/cloudwego/hertz/pkg/common/config"
"github.com/cloudwego/hertz/pkg/common/hlog"
```

**Comment by L2ncE** [comment]
I think you can open a new folder separately instead of directly changing the previous code.

**Comment by Skyenought** [comment]
可以用  `gofumpt -l -extra -w .` 整理一下代码格式

File: nacos/examples/custom_config/client/main.go [hunk 2]
```diff
@@ -53,12 +58,108 @@ func main() {
...
+
+	discoveryWithSD(r)
+	discoveryWithTag(r)
+	discoveryWithCustomizedAddr(r)
+	discoveryWithLoadBalanceOptions(r)
+	discoveryThenUsePostMethod(r)
+}
+
+func discoveryWithSD(r discovery.Resolver) {
+	fmt.Println("simply discovery:")
```
Comment (Skyenought) [inline nacos/examples/custom_config/client/main.go:70]: 这里全部的 fmt.Println 用 hlog.Info 吧

**Comment by dengWuuu** [comment]
> I think you can open a new folder separately instead of directly changing the previous code.

是要把以前的两个只启动一个hertz服务器的给从以前的复制回来？，然后新建两个文件夹？一个是启动两个hertz服务器的standard_multiple_server和custom_config_multiple_server?

File: nacos/examples/custom_config/client/main.go (hunk 2 above)
Comment (dengWuuu) [reply nacos/examples/custom_config/client/main.go]: done

File: nacos/examples/custom_config/client/main.go (hunk 1 above)
Comment (dengWuuu) [reply nacos/examples/custom_config/client/main.go]: done

**Comment by L2ncE** [comment]
> > I think you can open a new folder separately instead of directly changing the previous code.
> 
> 是要把以前的两个只启动一个hertz服务器的给从以前的复制回来？，然后新建两个文件夹？一个是启动两个hertz服务器的standard_multiple_server和custom_config_multiple_server?

原来的可以保留，新的文件夹根据你新增 example 的特性进行命名

**Review by Skyenought — APPROVED** [review]

**Comment by dengWuuu** [comment]
> > > I think you can open a new folder separately instead of directly changing the previous code.
> > 
> > 
> > 是要把以前的两个只启动一个hertz服务器的给从以前的复制回来？，然后新建两个文件夹？一个是启动两个hertz服务器的standard_multiple_server和custom_config_multiple_server?
> 
> 原来的可以保留，新的文件夹根据你新增 example 的特性进行命名

done

**Review by li-jin-gou — APPROVED** [review]

**Comment by li-jin-gou** [comment]
LGTM

**Pull request merged**